STYLE.md

```diff
@@ -0,0 +1,446 @@
+see many characters on one line.
+On the other hand, restricting line size to a very small number like 80 leads to
+some crazy indentation despite the fact that
+shorter lines should force you to write well structured code.
+That's why *100* is a reasonable compromize.
+
+
+### Indentation
+
+Tabs are illegal.  Use spaces for indenting.  Indent your code blocks
+with *2 spaces*.  Indent the `where` keyword with two spaces to set it
+apart from the rest of the code and indent the definitions in a
+`where` clause with 2 spaces.  Some examples:
+
+```haskell
+sayHello :: IO ()
+sayHello = do
+    name <- getLine
+    putStrLn $ greeting name
+  where
+    greeting name = "Hello, " ++ name ++ "!"
+
+filter :: (a -> Bool) -> [a] -> [a]
+filter _ [] = []
+filter p (x:xs)
+  | p x       = x : filter p xs
+  | otherwise = filter p xs
+```
+
+### Blank Lines
+
+* One blank line between top-level definitions.
+* No blank lines between type signatures and function definitions.
+* Add one blank line between
+functions in a type class instance declaration if the function bodies
+are large.
+* You can add blank lines inside a big `do` block to separate logical
+parts of it.
+* You can also use blank lines to separate definitions inside `where`
+clause, if necessary.
+
+### Whitespace
+
+* Surround binary operators with a single space on either side. In case of
+currying add one space between the argument and the operation.
+
+* Use some tool to remove trailing spaces.
+
+### Naming convention
+
+Casing:
++ **_lowerCamelCase_** for function and variable names.
++ **_UpperCamelCase_** for types.
+
+* Don't use short names like `n`, `sk`, `f` unless their meaning is clear from
+context (function name, types, other variables, etc.).
+
+* For readability reasons, don't capitalize all letters when using an
+abbreviation.  For example, write `HttpServer` instead of
+`HTTPServer`.  Exception: two or three letter abbreviations, e.g. `IO`, `STM`.
+
+**Records name conventions**
+
+If data type has only one constructor then this data type name should be same
+as constructor name (also applies to `newtype`).
+
+```haskell
+data User = User Int String
+```
+
+However when appropriate, some more context can be added to the constructor name.
+
+```haskell
+data Tx = UnsafeTx
+  { txInputs     :: !(NonEmpty TxIn)
+  -- ^ Inputs of transaction.
+  , txOutputs    :: !(NonEmpty TxOut)
+  -- ^ Outputs of transaction.
+  , txAttributes :: !TxAttributes
+  -- ^ Attributes of transaction
+  }
+```
+
+
+Convention for field names for record data types:
+
+Choose a reasonable prefix is used such that we can ensure the field name will remain unique throughout the package. Use your discretion.
+
+e.g
+```haskell
+data MempoolEnv m blk = MempoolEnv
+  { meEnvLedger    :: LedgerInterface m blk
+  , meEnvLedgerCfg :: LedgerConfig blk
+  , meEnvStateVar  :: StrictTVar m (InternalState blk)
+  , meEnvTracer    :: Tracer m (TraceEventMempool blk)
+  }
```

Comment:
If following Serge's point (https://github.com/input-output-hk/cardano-node/pull/164#discussion_r320014458), these field names would actually be `meLedger`, `meLedgerCfg`, `meStateVar`, `meTracer`, etc.